lib/elixir/lib/access.ex

```diff
@@ -195,8 +195,10 @@ defmodule Access do
       :error -> nil
     end
 
-    {get, update} = fun.(current_value)
-    {get, :maps.put(key, update, map)}
+    case fun.(current_value) do
+      {:skip, _} -> {:ok, map}
```

Comment:
This may be a hard problem to crack. Let me know if you want me to jump in and give it a try as well.